packages/payload/src/collections/operations/create.ts

```diff
@@ -110,8 +111,8 @@ async function create<TSlug extends keyof GeneratedTypes['collections']>(
     // /////////////////////////////////////
 
     const hasIdField =
-      collectionConfig.fields.findIndex((field) => fieldAffectsData(field) && field.name === 'id') >
-      -1
+      flattenFields(collectionConfig.fields).findIndex((field) => field.name === 'id') > -1
```

Comment:
It's relatively unfortunate that we have to `flattenFields` on every create, because it's a loop and is not necessary most of the time. Would be nice to, in the future, somehow store if there is a custom ID on a collection by collection basis on Payload itself, as the collections are being instantiated. Not necessary to change now, but something to keep on the radar.

Reply:
Added a @todo note in there, sounds like a good idea

Reply:
We should just write a `findField` utility function that works like flatten fields but returns as soon as it has a match. That will be more performent than flattening all fields, and looping again to find it.